Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -143,35 +143,50 @@ open class BasicFormat: SyntaxRewriter {
     return node.requiresIndent
   }
 
-  /// Whether a leading newline on `token` should be added.
-  open func requiresLeadingNewline(_ token: TokenSyntax) -> Bool {
-    // We don't want to add newlines inside string interpolation
-    if isInsideStringInterpolation(token) {
+  open func requiresNewline(between first: TokenSyntax?, and second: TokenSyntax?) -> Bool {
+    // We don't want to add newlines inside string interpolation.
+    // When first or second `TokenSyntax` is a multiline quote we want special handling
+    // event it's inside a string interpolation, because it still requires newline
+    // after open quote and before close quote.
+    if let first,
+      first.tokenKind != .multilineStringQuote,
+      second?.tokenKind != .multilineStringQuote,
+      isInsideStringInterpolation(first)
```

Comment:
I would swap the order of these checks because I think that way it reads the same way as the comment above.

Reply:
The reason I did this was of performance.

Reply:
Swapped it. 
If it should be another order I can do a follow up PR later.

Reply:
I was thinking about that as well but in >99% of all cases first and second won’t be multi-line string quotes, so we hit `isInsideStringInterpolation` basically all the time anyway.

Reply:
That was my thought. If `first` or `second` not multi line string, then is fails before is calls `isInsideStringInterpolation`